### PR TITLE
Add Health Check category group

### PR DESCRIPTION
**Pull request by andrepiske**
Adds the Health Check category group with a non-extensive list of projects. Addresses https://github.com/rubytoolbox/rubytoolbox/issues/1090

The goal is to list gems that provide some form of web/API endpoint(s) that perform health check (or liveness/readiness checks) by a third party. This is not about gems that call those endpoints.

**Before submitting your pull request:**

- [x] If you're referencing a gem by its GitHub repository (e.g. `rails/rails`), please verify
      that it is _not_ packaged as a Ruby gem. (If it _is_ packaged as a gem, please reference it
      by its gem name instead, e.g. `rails`.)
- [x] Please make sure the projects are listed in case-insensitive alphabetical order
- [ ] Make sure the CI build passes, we validate your proposed changes in the test suite :)

### CODE REVIEW

File: catalog/Maintenance_Monitoring/Health_Check.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Health Check
+description:
+  projects:
```
Comment (colszowka) [inline catalog/Maintenance_Monitoring/Health_Check.yml:3]: This indentation is wrong and needs to be top level, this currently makes the build fail

File: catalog/Maintenance_Monitoring/Health_Check.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: Health Check
```
Comment (colszowka) [inline catalog/Maintenance_Monitoring/Health_Check.yml:1]: We should go with plural "Health Checks" here please, i.e. we have "Web App Frameworks"; "JSON Parsers" and so on, and I suppose the gems themselves might also be capable of doing multiple checks depending on the use case so plural makes sense there as well

**Review by colszowka — COMMENTED** [review]
Thanks for contributing @andrepiske, please check out the 2 comments I have left, other than that good to go :+1:

**Comment by colszowka** [comment]
When in doubt, it would also be a nice to add a brief category description while we're at it, something like

> Easing the setup of uptime health checks for your applications that can be used for monitoring and deployment automation

**Comment by andrepiske** [comment]
@colszowka hey I've made the changes you requested! Just took me a few months 🙄

**Comment by colszowka** [comment]
Thanks @andrepiske, only a few months later this is now reviewed and I'm merging this, thanks for your contribution!

**Pull request merged**